src/client/dfs/dfs.c

```diff
@@ -914,31 +940,40 @@ create_dir(dfs_t *dfs, daos_handle_t parent_oh, daos_oclass_id_t cid,
 }
 
 static int
-open_dir(dfs_t *dfs, daos_handle_t th, daos_handle_t parent_oh, int flags,
+open_dir(dfs_t *dfs, daos_handle_t th, dfs_obj_t *parent, int flags,
 	 daos_oclass_id_t cid, struct dfs_entry *entry, size_t len,
 	 dfs_obj_t *dir)
 {
 	bool			exists;
 	int			daos_mode;
+	daos_handle_t		parent_oh;
 	int			rc;
 
+	parent_oh = parent ? parent->oh : dfs->super_oh;
+
 	if (flags & O_CREAT) {
-		rc = create_dir(dfs, parent_oh, cid, dir);
+		D_ASSERT(parent);
+
+		/** this generates the OID and opens the object */
+		rc = create_dir(dfs, parent, cid, dir);
 		if (rc)
 			return rc;
 
 		entry->oid = dir->oid;
 		entry->mode = dir->mode;
 		entry->atime = entry->mtime = entry->ctime = time(NULL);
-		entry->chunk_size = 0;
+		entry->chunk_size = parent->d.chunk_size;
+		entry->oclass = parent->d.oclass;
```

Comment:
This makes me wonder...
What if user creates a new directory with a given cid (for the directory itself) and wants the new directory to also use that cid as the default cid for children? Would they need to then call the set function?

Reply:
correct.
the oclass of the directory itself is totally separate from the oclass to be used for the children.